main.c

```diff
@@ -42,8 +44,16 @@ main(int argc, char **argv)
 
     ruby_sysinit(&argc, &argv);
     {
-	RUBY_INIT_STACK;
-	ruby_init();
-	return ruby_run_node(ruby_options(argc, argv));
+        RB_SETUP_EVENT_PROFILING_DEFAULT();
+        RB_SYSTEM_EVENT_PROFILING_BEGIN();
+
+        RUBY_INIT_STACK;
+        ruby_init();
+        int ret = ruby_run_node(ruby_options(argc, argv));
+
+        RB_SYSTEM_EVENT_PROFILING_END();
+        RB_FINALIZE_EVENT_PROFILING_DEFAULT();
+
+        return ret;
```

Comment:
Do these macros have to be placed in `main`, not `ruby_init`, `ruby_cleanup` and so on?

Reply:
I've moved them into `ruby.c:ruby_sysinit()` and `eval.c:rb_ec_cleanup()`